### ISSUE TITLE
Allow wildcards in hosts filter

### ISSUE THREAD

**Issue by LordMike** [ScottHelme/report-uri-bugs#1]
It's possible to set a wildcard hostname (*.example.com), but it's not explicitly allowed. Update documentation or function?

**Comment by ScottHelme** [comment]
Hi Mike, you're right. With the recent update you have to specify the hosts you want to collect reports for but specifying example.com will still collect reports for *.example.com too (I know this isn't indicated). I suppose there's no harm in allowing a wildcard subdomain as long as you're expecting reports for particular domain. I will put something together for this.

**Comment by ScottHelme** [comment]
I've pushed an update to the dev site with these changes, how does this sound?

![filters](https://cloud.githubusercontent.com/assets/5352840/14283149/edf8e38c-fb39-11e5-9227-9bb2b41cc337.png)

They will hit test later today and should be live this week if all goes well.

**Comment by LordMike** [comment]
Hmm.. this seems complex, but I'm beginning to see where you're headed.
Perhaps, trim down the options by applying sorting behind the scenes. So once you get the filter, parse it, then sort it so that more specific options come first, and then finally the wildcards.

If I enter `*.example.com www.example.com`, it should store: `www.example.com *.example.com`.

Another idea might also be to let the first suggestion be special (highlight it somehow), and mark it as "most common" or "this is what you're (most likely) looking for". For a first suggestion, I would put your no. 2: `*.example.com` and `example.com`.

**Comment by ScottHelme** [comment]
Trying to sort behind the scenes adds a greater level of complexity, especially when you consider I will have to deal with multiple levels of subdomain, *.test.example.com *.example.com example.com This also becomes even more complex when you involve multiple domains like *.test.domain.com *.domain.com domain.com

For now I'd like to get the changes deployed so those who need these features can use them and I can look at adding some intelligence around auto ordering later. All that is required at present is that the user correctly orders their entries.

**Comment by LordMike** [comment]
Wouldn't it be simple if you reversed it?

`moc.elpmaxe.*`
`moc.elpmaxe.www`

**Comment by LordMike** [comment]
Fun fact. If you think of DNS as the actual hierarchy it is (we all know it is - not many actually think of it like that), many things become possible. Before, I showed a pure string reversing, but if we parsed out the labels in the record requested, we could build it up like this:

* `com`
    * `example`
        * `www`
        * `*`
    * `domain`
        * `www`
        * `*`

This structure may store very little information (as we share common parts - such as `com.`).

Note: This has little to do with the actual problem.
